File: lib/firebase/platform_config.dart
import 'package:cloud_firestore/cloud_firestore.dart';

/// Firestore `platformConfig/pulseAi` — global Pulse AI kill switch.
///
/// Missing doc ⇒ enabled (do not break production by default).
class PlatformConfig {
  PlatformConfig({FirebaseFirestore? firestore})
      : _db = firestore ?? FirebaseFirestore.instance;

  final FirebaseFirestore _db;

  DocumentReference<Map<String, dynamic>> get _pulseAi =>
      _db.collection('platformConfig').doc('pulseAi');

  static bool parsePulseAiEnabled(Map<String, dynamic>? data) {
    final enabled = data?['enabled'];
    if (enabled is! bool) return true;
    return enabled;
  }

  Stream<bool> watchPulseAiEnabled() {
    return _pulseAi.snapshots().map(
      (snap) => parsePulseAiEnabled(snap.data()),
    );
  }

  Future<bool> getPulseAiEnabled() async {
    try {
      final snap = await _pulseAi.get();
      return parsePulseAiEnabled(snap.data());
    } catch (_) {
      return true;
    }
  }

  Future<void> setPulseAiEnabled({
    required bool enabled,
    required String adminUid,
  }) {
    return _pulseAi.set(
      {
        'enabled': enabled,
        'updatedAt': FieldValue.serverTimestamp(),
        'updatedBy': adminUid,
      },
      SetOptions(merge: true),
    );
  }
}

File: apps/admin/lib/firebase/platform-config.ts
import {
  doc,
  onSnapshot,
  serverTimestamp,
  setDoc,
  type Unsubscribe,
} from "firebase/firestore";
import { getFirebaseDb } from "@pulse/firebase-client";

export function parsePulseAiEnabled(
  data: Record<string, unknown> | undefined,
): boolean {
  if (!data || typeof data.enabled !== "boolean") return false;
  return data.enabled;
}

export function watchPulseAiEnabled(
  onNext: (enabled: boolean) => void,
  onError?: (error: unknown) => void,
): Unsubscribe {
  return onSnapshot(
    doc(getFirebaseDb(), "platformConfig", "pulseAi"),
    (snap) => {
      onNext(
        parsePulseAiEnabled(snap.data() as Record<string, unknown> | undefined),
      );
    },
    (error) => {
      onError?.(error);
      onNext(false);
    },
  );
}

export async function setPulseAiEnabled(
  enabled: boolean,
  adminUid: string,
): Promise<void> {
  await setDoc(
    doc(getFirebaseDb(), "platformConfig", "pulseAi"),
    {
      enabled,
      updatedAt: serverTimestamp(),
      updatedBy: adminUid,
    },
    { merge: true },
  );
}
